server/app/controllers/faq_controller.rb

```diff
@@ -1,44 +1,86 @@
 class FaqController < ApplicationController
   before_action :is_admin, except: [:show, :index]
 
-  def index
-    @faqs = Faq.all
-    render json: { faqs: faqs }
-  end
-
-  def show
-    @faq = Faq.find(params[:id])
-    render json: { faq: faq }
-  end
-
-  def create
-    faq = Faq.create(faq_params)
-    render json: { faq: faq }
-  end
-
-  def destroy
-    @faq = Faq.find(params[:id])
-    @faq.destroy
-    render json: {
-             message: "Faq deleted successfuly",
-           }, status: :ok
-  end
-
-  def update
-    @faq = Faq.find(params[:id])
-
-    if @faq.update(faq_params)
-      render json: {
-               message: "Faq updated successfuly",
-             }, status: :ok
-    else
-      render json: {
-               errors: ["Could not edit faq"],
-             }, status: :internal_server_error
+    def index
+        @faqs = Faq.all
+        render json: { faqs: @faqs }
     end
-  end
+  
 
-  def faq_params
-    params.require(:faq).permit(:question, :answer)
-  end
+    def show
+        @faq = Faq.find_by_id(params[:id])
+        if @faq
+          render json: {
+                   faq: @faq,
+                 }
+        else
+          render json: {
+                   status: 500,
+                   errors: ["Faq not found"],
+                 }
+        end
+      end
+
+    def find
+        uri = URI.parse(request.url)
+        params = CGI.parse(uri.query)
+        @faq = nil
+        puts params
+        if params.key?("question")
+            @faq = Faq.find_by_question(params["question"])
+        elsif params.key?("answer")
+            # Use this only for testing purposes as firstname-lastname pair is not guarenteed to be unique.
+            @faq = Faq.find_by answer: params["answer"]
+        end
+        if @faq
+            render json: {
+            faq: @faq,
+            }
+        else
+            render json: {
+                    status: 500,
+                    errors: ["faq not found"],
+                    }
```

Comment:
Could you change the way you return status per #22? It's a recent update so you might have missed it but that's how the front end expects the status code. It should be fairly easy.

Reply:
I looked at #22 but I didn't really get the message. I need some clarity